File: decomp/RandomGenerator.java
public class RandomGenerator {
   private int seed;

   public RandomGenerator(int var1) {
      this.seed = var1;
   }

   public void setSeed(int var1) {
      this.seed = var1;
   }

   public int nextInt() {
      this.seed = this.seed * 1593227 + 13;
      return this.seed >>> 16;
   }
}

File: original/js/randomGenerator.js
export class RandomGenerator {
  constructor(seed) {
    this.seed = seed | 0;
  }

  setSeed(seed) {
    this.seed = seed | 0;
  }

  nextInt() {
    // Must use 32-bit integer math to match Java's behavior
    this.seed = (Math.imul(this.seed, 1593227) + 13) | 0;
    return this.seed >>> 16;
  }
}
